src/Core/View.js

```diff
@@ -250,9 +250,10 @@ function _preprocessLayer(view, layer, provider) {
  * @param {Layer=} parentLayer
  * @return {Promise} a promise resolved with the new layer object when it is fully initialized
  */
+
 View.prototype.addLayer = function addLayer(layer, parentLayer) {
     const duplicate = this.getLayers((l => l.id == layer.id));
-    if (duplicate.length > 0) {
+    if (duplicate.length > 0 && !parentLayer) {
```

Comment:
The previous test was incorrect, but this one is wrong too.
I believe what we want is not instanciate 2 `Layer` objects with the same id.
So I'd suggest replace:
`&& !parentLayer` with `&& !(layer instanceof Layer)`.
WDYT?